nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/FlowController.java

```diff
@@ -311,7 +311,7 @@
     private final StandardFlowManager flowManager;
     private final RepositoryContextFactory repositoryContextFactory;
     private final RingBufferGarbageCollectionLog gcLog;
-    private final FlowEngine longRunningTaskMonitorThreadPool;
+    private final Optional<FlowEngine> longRunningTaskMonitorThreadPool;
```

Comment:
Although Optional is not meant to be used as a field type, this class is so voluminous that it makes sense to mark it here in case it won't be serialized at any time.

Reply:
In case of invalid time durations this is what I see from logs:

`2021-06-02 17:30:59,113 WARN [main] o.apache.nifi.controller.FlowController Could not retrieve value for nifi.monitor.long.running.task.schedule.
2021-06-02 17:30:59,114 WARN [main] o.apache.nifi.controller.FlowController Could not initialize LongRunningTaskMonitor.
java.lang.IllegalArgumentException: Value 'saf' is not a valid time duration`

I think as an enhancement we could provide the allowed time durations here too.

Reply:
I added some example values in the error error message.